jb/20210103/split_of_child_share_fixed.py

```diff
@@ -0,0 +1,362 @@
+  poverty_gap = pov_gap(spmu, 'new_spm_resources', 'spm_povthreshold', 'spm_weight')
+
+  # Merge person and spmu dataframes
+  spmu_sub = spmu[['spm_id', 'new_spm_resources', 'new_spm_resources_pp']]
+  target_persons = pd.merge(spmu_sub,person,on=['spm_id'])
+
+  target_persons['poor'] = target_persons.new_spm_resources < target_persons.spm_povthreshold
+  total_poor = (target_persons.poor * target_persons.person_weight).sum()
+  poverty_rate = (total_poor / pop * 100)
+
+  # Calculate Gini
+  gini = mdf.gini(target_persons, 'new_spm_resources_pp', w='person_weight')
+
+  # Percent winners
+  target_persons['better_off'] = (target_persons.new_spm_resources > target_persons.spm_resources)
+  total_better_off = (target_persons.better_off * target_persons.person_weight).sum()
+  percent_better_off = total_better_off / pop
+
+  return pd.Series([poverty_rate, gini, poverty_gap, percent_better_off, adult_ubi, child_ubi])
+
+# create a dataframe with all possible combinations of funding levels and
+summary = mdf.cartesian_product({'funding_billions': np.arange(0,3_001,50),
+                                 'child_percent_funding': np.arange(0, 101, 1)})
+
+def ubi_row(row):  
+    return ubi(row.funding_billions, row.child_percent_funding)
+summary[['poverty_rate', 'gini', 'poverty_gap', 'percent_better_off', 'adult_ubi', 'child_ubi']] = summary.apply(ubi_row, axis=1)
+
+summary['monthly_child_ubi'] =summary['child_ubi'].apply(lambda x: int(round(x/12,0)))
+summary['monthly_adult_ubi'] =summary['adult_ubi'].apply(lambda x: int(round(x/12,0)))
+summary.to_csv("children_share_funding_summary.csv.gz",compression='gzip')
+
+# drop rows where funding level is 0
+optimal_poverty = summary.sort_values('poverty_gap').drop_duplicates('funding_billions', keep='first')
+optimal_poverty = optimal_poverty.drop(
+    optimal_poverty[optimal_poverty.funding_billions==0].index
+    ) 
+
+optimal_inequality = summary.sort_values('gini').drop_duplicates('funding_billions', keep='first')
+optimal_inequality = optimal_inequality.drop(
+    optimal_inequality[optimal_inequality.funding_billions==0].index
+    ) 
+
+optimal_winners = summary.sort_values('percent_better_off').drop_duplicates('funding_billions', keep='last')
+optimal_winners = optimal_winners.drop(
+    optimal_winners[optimal_winners.funding_billions==0].index
+    )
+
+summary2 = mdf.cartesian_product({'funding_billions': np.arange(0,3_001,50),
+                                 'child_percent_ubi': np.arange(0, 101, 50)})
+
+def big_percent_row(row):  
+    return ubi2(row.funding_billions, row.child_percent_ubi)
+
+summary2[['poverty_rate', 'gini', 'poverty_gap', 'percent_better_off', 'adult_ubi', 'child_ubi']] = summary2.apply(big_percent_row, axis=1)
+summary2
+
+# calculate monthly payments
+summary2['monthly_child_ubi'] =summary2['child_ubi'].apply(lambda x: int(round(x/12,0)))
+summary2['monthly_adult_ubi'] =summary2['adult_ubi'].apply(lambda x: int(round(x/12,0)))
+summary2.sample(10)
```

Comment:
remove this and any other code that doesn't produce the result dataframe(s) used in the notebook (e.g. also graphing code)

Reply:
keeping the monthly, calculation, got rid of line 158